kernel/kernel-impl/src/main/java/org/sakaiproject/user/impl/FallbackDetailProvider.java

```diff
@@ -0,0 +1,102 @@
+package org.sakaiproject.user.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.sakaiproject.component.api.ServerConfigurationService;
+import org.sakaiproject.component.cover.ComponentManager;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.api.SiteService;
+import org.sakaiproject.tool.api.ToolManager;
+import org.sakaiproject.user.api.CandidateDetailProvider;
+import org.sakaiproject.user.api.User;
+import org.sakaiproject.util.api.ValueEncryptionService;
+
+public class FallbackDetailProvider implements CandidateDetailProvider {
+	
+	private static Log M_log = LogFactory.getLog(FallbackDetailProvider.class);
+	
+	private ServerConfigurationService serverConfigurationService;
+	private SiteService siteService;
+	private ToolManager toolManager;
+	private ValueEncryptionService valueEncryptionService;
+	
+	public void init() {
+		if(valueEncryptionService == null){
```

Comment:
We though it was a good idea to let a custom service injection via spring (for example for a new valueExncryptionService) and also let a default service initialisation if there is not any injection.

Anyway we are going to update it with Objects.requireNonNull